examples/viewer-integr-messaging.js

```diff
@@ -47,11 +47,13 @@ function ViewerMessaging(target, targetOrigin, requestProcessor, opt_targetId) {
   /** @private {function(string, *, boolean):(!Promise<*>|undefined)} */
   this.requestProcessor_ = requestProcessor;
 
-  if (!this.targetOrigin_) {
-    throw new Error('Target origin must be specified');
-  }
+  this.pipe_ = opt_pipe;
 
-  window.addEventListener('message', this.onMessage_.bind(this), false);
+  if (opt_pipe) {
+    opt_pipe.addEventListener('message', this.onMessage_.bind(this));
```

Comment:
this specific one is a mocked port so it doesn't need start, but for making both sets of code equal done :)